Source/WebCore/rendering/GridMasonryLayout.cpp

```diff
@@ -154,13 +154,39 @@ GridArea GridMasonryLayout::gridAreaForDefiniteGridAxisItem(const RenderBox& gri
     return masonryGridAreaFromGridAxisSpan(itemSpan);
 }
 
-void GridMasonryLayout::placeItemsWithIndefiniteGridAxisPosition(const GridTrackSizingAlgorithm& algorithm)
+void GridMasonryLayout::placeItemsWithIndefiniteGridAxisPosition(const GridTrackSizingAlgorithm& algorithm, GridMasonryLayout::OverrideLogicalWidth overrideLogicalWidthType)
 {
     for (auto* item : m_itemsWithIndefiniteGridAxisPosition) {
         ASSERT(item);
         if (!item)
             continue;
-        insertIntoGridAndLayoutItem(algorithm, *item, gridAreaForIndefiniteGridAxisItem(*item));
+        insertIntoGridAndLayoutItem(algorithm, *item, gridAreaForIndefiniteGridAxisItem(*item), overrideLogicalWidthType);
+    }
+}
+
+void GridMasonryLayout::overrideLogicalWidth(RenderBox& gridItem, GridMasonryLayout::OverrideLogicalWidth overrideLogicalWidthType)
+{
+    switch (overrideLogicalWidthType) {
+    case OverrideLogicalWidth::DoNotOverride:
+        break;
+    case OverrideLogicalWidth::MinContent:
+        if ((gridItem.style().logicalWidth().isAuto() || gridItem.style().logicalWidth().isPercent()) && !(gridItem.isOutOfFlowPositioned() || gridItem.isExcludedFromNormalLayout())) {
```

Comment:
Can we not continue to reuse the order iterator logic for skipping grid items here? Seems like some of the other masonry code also uses it

Reply:
These extra checks are not needed as we perform the checks earlier. Removed.